examples/konnect-kic-lambda-plugin/main.tf

```diff
@@ -0,0 +1,266 @@
+  statement {
+    sid    = "KongPodPolicy"
+    effect = "Allow"
+
+    actions = [
+      "lambda:InvokeFunction"
+    ]
+
+    resources = [resource.aws_lambda_function.konnect_plugin_demo_lambda.arn]
+  }
+}
+
+
+################################################################################
+# Kubernetes Service Account for Kong Pod - Not required here as its moved to Kong module values
+################################################################################
+
+# resource "kubernetes_service_account" "kong_pod_service_account" {
+#   metadata {
+#     name      = local.kong_pod_service_account_name
+#     namespace = module.eks_blueprints_kubernetes_addon_kong.namespace
+#     annotations = {
+#       "eks.amazonaws.com/role-arn" = module.iam_assumable_role_admin.iam_role_arn
+#     }
+#   }
+# }
+
+################################################################################
+# Sample Lambda Hello World function in nodejs
+################################################################################
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+
+}
+
+resource "aws_iam_role" "iam_for_lambda" {
+  name               = "konnect-plugin-demo-lambda"
+  assume_role_policy = data.aws_iam_policy_document.assume_role.json
+}
+
+data "archive_file" "lambda" {
+
+  type        = "zip"
+  output_path = "${path.root}/.archive_files/lambda_function_payload.zip"
+
+  # fingerprinter
+  source {
+    filename = "index.js"
+    content  = <<CODE
+      exports.handler=async t=>{let e={statusCode:200,body:JSON.stringify("Hello from Lambda!")};return e};
+      CODE
+  }
+
+}
+
+resource "aws_lambda_function" "konnect_plugin_demo_lambda" {
+
+  filename         = "${path.root}/.archive_files/lambda_function_payload.zip"
+  function_name    = "konnect-plugin-demo-lambda"
+  description      = "Hello World Lambda Function"
+  handler          = "index.handler"
+  source_code_hash = data.archive_file.lambda.output_base64sha256
+
+  runtime = "nodejs18.x"
+  role    = aws_iam_role.iam_for_lambda.arn
+
+}
+
+###############################################################################
+#CRD to create Kong Lambda Plugin
+###############################################################################
+resource "kubernetes_manifest" "kong_lambda_plugin" {
+  depends_on = [ module.eks_blueprints_kubernetes_addon_kong,module.eks ]
+  manifest = {
+    "apiVersion" = "configuration.konghq.com/v1"
+    "kind"       = "KongPlugin"
+    "metadata" = {
+      "name"      = "konnect-plugin-demo-lambda"
+      "namespace" = local.kong_namespace
+    }
+    "plugin" = "aws-lambda"
+    "config" = {
+      "function_name" = resource.aws_lambda_function.konnect_plugin_demo_lambda.function_name
+      "forward_request_method" = true
```

Comment:
Can we add here `awsgateway_compatible: true` which sets the plugin to wrap requests into AWS API gateway format? Or at least have it commented out and have a comment explaining what it does? Since this is using Lambda, the user will probably want it to be compatible with the API Gateway [payload format version](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html).

Same with `is_proxy_integration: true` which sets the response from Lambda so that only the body will be returned as the response, which is what API Gateway does.